### PR TITLE
fix(react): update handleSize function on the useBagItem

### PR DESCRIPTION
**Pull request by catarinacouto**
## Description

This prevents the handleSize function from updating a size and adding the same size after. This happens because when
the previous bag quantity is bigger than the merchant quantity for the size change, the function `handleAddOrUpdateBagItem` is called with all the merchants of the size, instead of being called with all the merchants, except the one that is updated previously.

### Dependencies

None.

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Review by anapim — APPROVED** [review]

**Comment by helderburato** [comment]
@boliveira could you give a check to this PR? Something is broken in the pipeline.

**Comment by boliveira** [comment]
> @boliveira could you give a check to this PR? Something is broken in the pipeline.

This needs a rebase with `next` to catch the latest changes to the pipeline that include the fix.

File: packages/react/src/bags/hooks/useBagItem.ts [hunk 1]
```diff
@@ -39,7 +40,7 @@ import type { StoreState } from '@farfetch/blackout-redux/src/types';
  *
  * @param {number} [bagItemId] - Bag item id to work on. Optional only if you need
  * to use the `handleAddOrUpdateItem` method. For more information and/or future plans
- * check the [respective issue](https://github.com/Farfetch/blackout/issues/16).
+ * check the [respective issue](https://gitlab.fftech.info/fps/platform/box/issues/160).
```
Comment (nelsonleite) [inline packages/react/src/bags/hooks/useBagItem.ts:43]: Please revert this change

**Review by nelsonleite — CHANGES_REQUESTED** [review]

File: packages/react/src/bags/hooks/useBagItem.ts (hunk 1 above)
Comment (catarinacouto) [reply packages/react/src/bags/hooks/useBagItem.ts]: Applied ✅

**Review by nelsonleite — APPROVED** [review]

**Review by helderburato — APPROVED** [review]

**Review by gabrielfmp — APPROVED** [review]

**Pull request merged**